File: backend/app/schemas/market.py
from datetime import datetime

from pydantic import BaseModel, Field


class MarketSnapshot(BaseModel):
    symbol: str
    source: str
    timestamp: datetime
    bid: float
    ask: float
    spread: float
    spread_points: int
    digits: int
    data_status: str
    is_mock: bool
    is_tradable: bool
    note: str = Field(
        default="Mock market data for development only. Not a trading signal."
    )

File: frontend/src/api/market.ts
import { apiGet } from "./client";

export type MarketSnapshot = {
  symbol: string;
  source: string;
  timestamp: string;
  bid: number;
  ask: number;
  spread: number;
  spread_points: number;
  digits: number;
  data_status: string;
  is_mock: boolean;
  is_tradable: boolean;
  note: string;
};

export function fetchMarketSnapshot(): Promise<MarketSnapshot> {
  return apiGet<MarketSnapshot>("/api/market/snapshot");
}
